cobigen-cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java

```diff
@@ -317,13 +323,40 @@ public void generate(File inputFile, File inputProject, List<? extends Generable
                 logger.info("Generating increments for input '" + inputFile.getName() + "', this can take a while...");
                 report = cg.generate(input, finalTos, Paths.get(outputRootPath.getAbsolutePath()), false, utilClasses);
             }
-            ValidationUtils.checkGenerationReport(report);
+            if (isJavaInput && ValidationUtils.checkGenerationReport(report)) {
+                Set<Path> format = report.getGeneratedFiles();
+                Iterator<Path> itr = format.iterator();
+                if (itr.hasNext()) {
+                    try {
+
+                        try {
+                            InputStream inputStream = Files.newInputStream(itr.next());
+                            String unformatterContent = IOUtils.toString(inputStream);
+                            new Formatter().formatSource(unformatterContent);
```

Comment:
You are creating a new `Formatter` for each loop iteration. Please move the declaration before the iteration and reuse the same instance.

Also here you are formatting the code but you are not replacing the old contents of the file. You need to remove the old file contents and replace them with the formatted string